### PR TITLE
Updated codebase to ES6

### PR DESCRIPTION
**Pull request by dgreene-r7**
Also introduced eslint and rapid7 ruleset and resolved linting issues.

### CODE REVIEW

**Comment by fmitchell-r7** [comment]
Thanks for putting this together!

When I run with debugging turned on (`npm run debug`) there's a new warning that shows up in the console log. I'm not sure where it's coming from or what it means.
```
Warning: Failed propType: Required prop `error` was not specified in `Configure`.
```

File: lib/aws-credentials.js [hunk 1]
```diff
@@ -1,66 +1,68 @@
...
+        let config = Object.create(null);
+
+        if (data && data !== '') {
+          config = ini.parse(data);
+        }
+
+        config[profile] = {};
+        config[profile].aws_access_key_id = credentials.AccessKeyId;
+        config[profile].aws_secret_access_key = credentials.SecretAccessKey;
+
```
Comment (fmitchell-r7) [inline lib/aws-credentials.js]: It looks like the `aws_session_token` value is missing. It should be set the same as `aws_security_token`.

**Comment by dgreene-r7** [comment]
Good catch. It's coming out of the `views/configure.jsx` file and it's happening because of the block from [Lines 6-10](https://github.com/dgreene-r7/awsaml/blob/es6-compat/views/configure.jsx#L6-L10). The `propType` for `error` should be `React.PropTypes.bool` rather than `React.PropTypes.bool.isRequired`. Next commit will fix this.

File: lib/aws-credentials.js (hunk 1 above)
Comment (dgreene-r7) [inline lib/aws-credentials.js]: I noticed that and thought it was a copy pasta thing. Will fix.

**Pull request merged**